### PR TITLE
[WIP]Replaced Npm with Yarn in setup-services.sh

### LINKED ISSUE #1925: Consider using yarn rather than npm for this project
This might also solve the binlink NPM run issue https://github.com/mozilla/thimble.mozilla.org/pull/1714

### PR DESCRIPTION
**Pull request by JohnEJames**
Fixes #1925 
I am not able to test this because both my machines seem to be denying port 3500 when I try to vagrant up. I Thought I'd put up the pull request for feedback. I'll keep trying to get thimble to work on my local machine, but I am not having much luck

Sorry about this in advance

### CODE REVIEW

**Comment by gideonthomas** [comment]
@Pomax any chance you can test this on windows?

File: scripts/setup-services.sh [hunk 1]
```diff
@@ -24,15 +24,19 @@ apt-get update -y
...
 update-alternatives --install /usr/bin/gcc gcc /usr/bin/gcc-4.8 60 --slave /usr/bin/g++ g++ /usr/bin/g++-4.8
 # ---
+# -- Installing Yarn
+curl -sS https://dl.yarnpkg.com/debian/pubkey.gpg | sudo apt-key add -
+echo "deb https://dl.yarnpkg.com/debian/ stable main" | sudo tee /etc/apt/sources.list.d/yarn.list
+#--
 apt-get install -y git # For id.webmaker.org
-npm install -g http-server --loglevel=error # For hosting published projects
-npm install -g node-gyp --loglevel=error # For bcrypt
+yarn global add http-server --loglevel=error # For hosting published projects
```
Comment (humphd) [inline scripts/setup-services.sh]: Does `yarn` support `--loglevel=error`?

File: scripts/setup-services.sh [hunk 2]
```diff
@@ -83,15 +87,15 @@ cd services
 echo "Setting up login.webmaker.org"
 cd login.webmaker.org
 cp env.sample .env
-sudo npm install --no-bin-links --loglevel=error # sudo needed for bcrypt permissions
+sudo yarn install --no-bin-links --loglevel=error # sudo needed for bcrypt permissions
```
Comment (humphd) [inline scripts/setup-services.sh]: Same question here, `--no-bin-links`, is that supported with `yarn`?

File: scripts/setup-services.sh [hunk 3]
```diff
@@ -24,15 +24,19 @@ apt-get update -y
...
-npm install -g node-pre-gyp --loglevel=error # For login.webmaker.org sqlite3
-npm install -g autoless@0.1.7 webpack@1.7.3 --loglevel=error # For id.webmaker.org
-npm install -g knex@">=0.8.6 <0.9.0" --loglevel=error # For publish.webmaker.org
-npm install -g pm2 --loglevel=error # For running Thimble, see scripts/start-services.sh
-npm install -g grunt-cli --loglevel=error # For Thimble's npm postinstall script
+yarn global add node-pre-gyp --loglevel=error # For login.webmaker.org sqlite3
+yarn global add autoless@0.1.7 webpack@1.7.3 --loglevel=error # For id.webmaker.org
+yarn global add knex@">=0.8.6 <0.9.0" --loglevel=error # For publish.webmaker.org
+yarn global add pm2 --loglevel=error # For running Thimble, see scripts/start-services.sh
+yarn global add grunt-cli --loglevel=error # For Thimble's npm postinstall script
```
Comment (humphd) [inline scripts/setup-services.sh]: You use both `add` and `install`.  Is there a reason for both?

Comment (JohnEJames) [inline scripts/setup-services.sh]: In the Document you gave me https://yarnpkg.com/lang/en/docs/migrating-from-npm/. it said for npm install global, use the command add. and npm install is just yarn install

File: scripts/setup-services.sh (hunk 2 above)
Comment (JohnEJames) [inline scripts/setup-services.sh]: From what I read from here https://github.com/yarnpkg/yarn/issues/1961 it looks like it is

File: scripts/setup-services.sh (hunk 1 above)
Comment (JohnEJames) [inline scripts/setup-services.sh]: for this it looks like the command is yarn run
https://github.com/yarnpkg/yarn/issues/1523

**Comment by ryanwarsaw** [comment]
https://github.com/yarnpkg/yarn/pull/3097

We may need to wait for this to land, its relevant to ```--no-bin-links``` flag (see above).

**Comment by gideonthomas** [comment]
@ryanwarsaw how so? `no-bin-links` tells npm not to use symlinks whereas that issue has to do with failures while creating symlinks. Am I missing something?

**Comment by ryanwarsaw** [comment]
@gideonthomas I think it was just a derp on my part, I'll have to go back later and read through everything and make sure I didn't just accidentally post the wrong link, my bad.

**Comment by JohnEJames** [comment]
Hey sorry, I haven't been here in awhile, next week I should be able to start working on this pull request!

**Comment by JohnEJames** [comment]
I have finally fixed my vagrant issue, so I can test this PR now.

**Comment by humphd** [comment]
This fails for me like so:

```
==> default: Processing triggers for libc-bin (2.19-0ubuntu6.11) ...
==> default: update-alternatives:
==> default: using /usr/bin/gcc-4.8 to provide /usr/bin/gcc (gcc) in auto mode
==> default: OK
==> default: deb https://dl.yarnpkg.com/debian/ stable main
==> default: Reading package lists...
==> default: Building dependency tree...
==> default: Reading state information...
==> default: The following extra packages will be installed:
==> default:   git-man liberror-perl
==> default: Suggested packages:
==> default:   git-daemon-run git-daemon-sysvinit git-doc git-el git-email git-gui gitk
==> default:   gitweb git-arch git-bzr git-cvs git-mediawiki git-svn
==> default: The following NEW packages will be installed:
==> default:   git git-man liberror-perl
==> default: 0 upgraded, 3 newly installed, 0 to remove and 7 not upgraded.
==> default: Need to get 3,361 kB of archives.
==> default: After this operation, 21.9 MB of additional disk space will be used.
==> default: Get:1 http://archive.ubuntu.com/ubuntu/ trusty/main liberror-perl all 0.17-1.1 [21.1 kB]
==> default: Get:2 http://archive.ubuntu.com/ubuntu/ trusty-updates/main git-man all 1:1.9.1-1ubuntu0.5 [699 kB]
==> default: Get:3 http://archive.ubuntu.com/ubuntu/ trusty-updates/main git amd64 1:1.9.1-1ubuntu0.5 [2,641 kB]
==> default: dpkg-preconfigure: unable to re-open stdin: No such file or directory
==> default: Fetched 3,361 kB in 5s (656 kB/s)
==> default: Selecting previously unselected package liberror-perl.
==> default: (Reading database ... 68176 files and directories currently installed.)
==> default: Preparing to unpack .../liberror-perl_0.17-1.1_all.deb ...
==> default: Unpacking liberror-perl (0.17-1.1) ...
==> default: Selecting previously unselected package git-man.
==> default: Preparing to unpack .../git-man_1%3a1.9.1-1ubuntu0.5_all.deb ...
==> default: Unpacking git-man (1:1.9.1-1ubuntu0.5) ...
==> default: Selecting previously unselected package git.
==> default: Preparing to unpack .../git_1%3a1.9.1-1ubuntu0.5_amd64.deb ...
==> default: Unpacking git (1:1.9.1-1ubuntu0.5) ...
==> default: Processing triggers for man-db (2.6.7.1-1ubuntu1) ...
==> default: Setting up liberror-perl (0.17-1.1) ...
==> default: Setting up git-man (1:1.9.1-1ubuntu0.5) ...
==> default: Setting up git (1:1.9.1-1ubuntu0.5) ...
==> default: /tmp/vagrant-shell: line 32: yarn: command not found
The SSH command responded with a non-zero exit status. Vagrant
assumes that this means the command failed. The output for this command
should be in the log above. Please read the output to determine what
went wrong.
```

@gideonthomas I wonder if we should close this, and push on https://github.com/mozilla/thimble.mozilla.org/pull/1714 instead to get `npm@5`?

**Pull request closed without merging**